tests/test_tasks.py

```diff
@@ -2978,6 +2979,34 @@ def my_flow():
         task_states = my_flow()
         assert [state.result() for state in task_states] == [6, 7, 8]
 
+    def test_map_with_sequential_runner_is_sequential(self):
+        """Tests that the sequential runner executes mapped tasks sequentially. The first tasks
+        sleeps for 3 seconds before appending and the second for 0, so if the tasks do not execute
+        sequentially, we expect the second task to append before the first.
+        """
+
+        @task
+        def sleepy_task(n, mock_item):
+            time.sleep(n)
+            mock_item(n)
+            return n
+
+        @flow
+        def my_flow(mock_item):
+            sleepy_task.map([3, 0], unmapped(mock_item))
+
+        mock_item = MagicMock()
```

Comment:
Note you could just be appending integers to a list instead of calling a mock.

Reply:
I tried this and couldn't make it work for some reason